Fix 930 pyreverse regression

DudeNr33 opened the pull request:
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
PR #857 introduced a regression for ``modutils.is_relative()``.
Calling this function failed if the inspected import statement was inside a module of a package at least two levels deep (e.g. ``astroid.interpreter._import.spec``, see the added unit tests).

Inspecting the old implementation of ``is_relative`` which relied on ``imp`` and the implementation of ``imp.find_module``, this only checked of the given ``modname`` (or better to say the first part after splitting it on ".") was a module or package inside the same package as the ``from_file`` itself. 

The new implementation seemed to do a lot more, and I could not really understand what exactly.

Anyways, ``importlib.machinery.PathFinder().find_spec`` [seems to be a better replacement for ``imp.find_module``](https://stackoverflow.com/a/50028745/4030694), makes the implementation simpler, passes all the tests and ``pyreverse`` is working again.

## Type of Changes

<!-- Leave the corresponding lines for the applicable type of change: -->

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

Closes #930 
Closes PyCQA/pylint#4186

DudeNr33 commented:
@Pierre-Sassoulas or @cdce8p could you please approve the workflow run? 
I fixed the pre-commit checks that were failing on the previous run.

hippo91 commented on astroid/modutils.py:
```diff
@@ -574,21 +574,12 @@ def is_relative(modname, from_file):
-    submodule_spec = importlib.util.find_spec(
-        name + "." + modname.split(".")[0], parent_spec.submodule_search_locations
+    modspec = importlib.machinery.PathFinder().find_spec(
+        modname.split(".")[0], [from_file]
```
From what i can read in the `PathFinder` doc, it seems to do the job. However i would like to have @pkolbus opinion on this. Maybe he could tell us if did not use the `PathFInder` for a specific reason. 
Instead of `split` you can use `partition`. Performance is a bit better as it doesn't split multiple times the string but just one. Moreover, as the `PathFinder` doc say, there is no need to instantiate this class as all its methods are `@classmethod`. 
We thus could directly write:
```
return importlib.machinery.PathFinder.find_spec(modname.partition(".")[0], [from_file])
```

pkolbus replied:
This part of #857 wasn’t my work actually, it came from @degustaf in #686. I didn’t run into any errors with it, so didn’t evaluate alternatives. @degustaf, can you provide insight here?

hippo91 replied:
> This part of #857 wasn’t my work actually, it came from @degustaf in #686. I didn’t run into any errors with it, so didn’t evaluate alternatives. @degustaf, can you provide insight here?

Thanks @pkolbus !

hippo91 reviewed (approved):
@DudeNr33 thanks for this clear, concise and tested MR!
Maybe @cdce8p could also give his opinion on this.
Before merging it to master i would also like to have @pkolbus (author of #857) opinion on the use of `PathFinder` class.

cdce8p reviewed (changes requested):
I noticed that `astroid/modutils.py` seems to be missing some imports
```py
import importlib
import importlib.machinery
```

DudeNr33 commented:
> I noticed that `astroid/modutils.py` seems to be missing some imports
> 
> ```python
> import importlib
> import importlib.machinery
> ```

Added the imports and removed the instantiation of the PathFinder class.
Interesting that it already worked without the imports - and while PyCharm actually complained about the imports, VS Code did not.

cdce8p commented on astroid/modutils.py:
```diff
@@ -574,21 +577,9 @@ def is_relative(modname, from_file):
-    submodule_spec = importlib.util.find_spec(
-        name + "." + modname.split(".")[0], parent_spec.submodule_search_locations
+    return bool(
+        importlib.machinery.PathFinder.find_spec(modname.partition(".")[0], [from_file])
```
```suggestion
        importlib.machinery.PathFinder.find_spec(modname.split(".", maxsplit=1)[0], [from_file])
```
We started to use `split` with `maxsplit=1` instead for pylint. Might be an idea to use here as well.

DudeNr33 replied:
Changed it, thanks for the suggestion.

cdce8p commented on ChangeLog:
```diff
@@ -50,7 +54,6 @@ Release Date: TBA
 
   Closes #485
 
```
```suggestion


```
